modules/Components/src/client/ACASThingBrowser.coffee

```diff
@@ -69,16 +63,48 @@ class ThingSimpleSearchController extends AbstractFormController
 					values: [
 
 					]
-
+				returnDTO: {
+					thingValues: []
+					thingAttributes: ["id", "codeName"]
+				}
+									
 			queryTerms = @getQueryTerms(defaultQueryTerms, thingSearchTerm)
+
+			for queryValue in @configs
+				valDef = @model.getValueInfo(queryValue.key)
+
+				# Anything that is in the reserved words list is not a value but rather an lsthing attribute so we just add it to the thingAttributes list
+				if queryValue.key in @thingAttributeReservedWords
+					# Only add it to the thingattributes if it is not already there
+					if queryValue.key not in queryTerms.returnDTO.thingAttributes
+						queryTerms.returnDTO.thingAttributes.push(queryValue.key)
+					continue
+
+				if valDef?
+					queryTerms.returnDTO.thingValues.push	
+						stateType: valDef.stateType
+						stateKind: valDef.stateKind
+						valueType: valDef.type
+						valueKind: valDef.kind
+						key: queryValue.key
+				else
+					# If the key is an ls label
+					labDef = @model.getLabelInfo(queryValue.key)
+					if labDef?
+						queryTerms.returnDTO.thingValues.push	
+							labelType: labDef.type
+							labelKind: labDef.kind
+							key: queryValue.key
+
+
 			$.ajax
 				type: 'POST'
-				url: "#{@genericSearchUrl}#{@model.get("lsType")}/#{@model.get("lsKind")}?format=nestedfull"
+				url: "#{@genericSearchUrl}#{@model.get("lsType")}/#{@model.get("lsKind")}?format=flat"
 				dataType: 'json',
 				contentType: 'application/json'
 				data: JSON.stringify(queryTerms)
-				success: (thing) =>
-					@trigger "searchReturned", thing.results
+				success: (response) =>
+					@trigger "searchReturned", response
```

Comment:
Instead of only passing the "results" we pass the full object so we can use the maxResults and numberOfResults to display a message to the user about hitting the limit of results.